device_service.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+)
+
+type Device struct {
+	Id       int     `json:"id"`
+	Name     string  `json:"name"`
+	Interval int     `json:"interval"`
+	Value    float32 `json:"value"`
+}
+
+type DeviceService struct {
+	Database
+}
+
+func (service *DeviceService) createDevice(w http.ResponseWriter, r *http.Request) {
+	var device Device
+
+	if err := json.NewDecoder(r.Body).Decode(&device); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	if err := service.validate(device); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	device, err := service.saveDevice(device)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
It is good to just log the error message from DAO and return something more generic. We don't want to publish information about our internals to the end user.